packages/typedoc-plugin-appium/lib/converter/converter.ts

```diff
@@ -0,0 +1,417 @@
+/**
+ * Name of the command in a method map
+ */
+export const NAME_COMMAND = 'command';
+
+/**
+ * Name of the field in a _regular_ method map which contains parameters
+ */
+export const NAME_PAYLOAD_PARAMS = 'payloadParams';
+
+/**
+ * Name of the module which contains the builtin method map
+ */
+export const NAME_BUILTIN_COMMAND_MODULE = '@appium/base-driver';
+
+/**
+ * Converts declarations to information about Appium commands
+ */
+export class CommandConverter {
+  #ctx: Context;
+  #log: AppiumPluginLogger;
+
+  /**
+   * Creates a child logger for this instance
+   * @param ctx Typedoc Context
+   * @param log Logger
+   */
+  constructor(ctx: Context, log: AppiumPluginLogger) {
+    this.#ctx = ctx;
+    this.#log = log.createChildLogger('converter');
+  }
+
+  /**
+   * Converts declarations into command information
+   *
+   * @returns Command info for entire project
+   */
+  public convert(): ModuleCommands {
+    const ctx = this.#ctx;
+    const {project} = ctx;
+    const projectCommands: ModuleCommands = new Map();
+
+    // handle baseDriver if it's present
+    const baseDriver = project.getChildByName(NAME_BUILTIN_COMMAND_MODULE);
+    if (baseDriver && isBaseDriverDeclarationReflection(baseDriver)) {
+      this.#log.verbose('Found %s', NAME_BUILTIN_COMMAND_MODULE);
+      projectCommands.set(baseDriver, this.#convertBaseDriver(baseDriver));
+    } else {
+      this.#log.verbose('Did not find %s', NAME_BUILTIN_COMMAND_MODULE);
+    }
+
+    // convert all modules (or just project if no modules)
+    const modules = project.getChildrenByKind(ReflectionKind.Module);
+    if (modules.length) {
+      for (const mod of modules) {
+        this.#log.verbose('Converting module %s', mod.name);
+        const cmdInfo = this.#convertModuleClasses(mod);
+        if (cmdInfo.hasCommands) {
+          projectCommands.set(mod, this.#convertModuleClasses(mod));
+        }
+        this.#log.info('Converted module %s', mod.name);
+      }
+    } else {
+      projectCommands.set(project, this.#convertModuleClasses(project));
+    }
+
+    this.#log.info('Found commands in %d module(s)', projectCommands.size);
+
+    return projectCommands;
+  }
+
+  #convertBaseDriver(baseDriver: BaseDriverDeclarationReflection): CommandInfo {
+    const baseDriverRoutes = this.#convertMethodMap(baseDriver);
+    if (!baseDriverRoutes.size) {
+      throw new TypeError(`Could not find any commands in BaseDriver!?`);
+    }
+
+    // no execute commands in BaseDriver
+    return new CommandInfo(baseDriverRoutes);
+  }
+
+  /**
+   * Finds names of parameters of a command in a method def
+   * @param propName Either required or optional params
+   * @param refl Parent reflection (`params` prop of method def)
+   * @returns List of parameter names
+   */
+
+  #convertCommandParams(
+    propName: typeof NAME_OPTIONAL | typeof NAME_REQUIRED,
+    refl?: DeclarationReflectionWithReflectedType
+  ): string[] {
+    if (refl) {
```

Comment:
I would rather flip the condition and make it to a guard to avoid long nested code blocks